scripts/bulk-editor/tag-command.ts

```diff
@@ -0,0 +1,186 @@
+/* This file is a part of @mdn/browser-compat-data
+ * See LICENSE file for more information. */
+
+import fsp from 'node:fs/promises';
+import path from 'node:path';
+import fs from 'node:fs';
+
+import chalk from 'chalk-template';
+
+/**
+ * isObject - Helper function testing for an object
+ * @param item Value to test
+ * @returns A boolean saying if it is an object or not
+ */
+const isObject = (item: any) =>
+  item && typeof item === 'object' && !Array.isArray(item);
+
+/**
+ * mergeDeep - Merge several JSON object
+ * @param target The first JSON object
+ * @param sources The JSON object to merge to the first one
+ * @returns The merged JSON object
+ */
+const mergeDeep = (target: object, ...sources: object[]) => {
+  // Assume there is no circular references (or this leads to an infinite loop)
+
+  if (!sources.length) {
+    return target;
+  }
+
+  const source = sources.shift();
+
+  if (isObject(target) && isObject(source)) {
+    for (const key in source) {
+      if (isObject(source[key])) {
+        if (!target[key]) {
+          Object.assign(target, { [key]: {} });
+        }
+        mergeDeep(target[key], source[key]);
+      } else {
+        Object.assign(target, { [key]: source[key] });
+      }
+    }
+  }
+
+  return mergeDeep(target, ...sources);
+};
+
+/**
+ * bcdAssociatedFilename - Generate the filename containing the bcd ID
+ * @param bcdID The bcd ID we are looking for
+ * @param path The path to the JSON files
+ * @returns A string with the filename
+ */
+const bcdAssociatedFilename = (bcdID: string, path: string): string => {
```

Comment:
I wonder if this is really needed? From the call sites it looks like this could potentially return a new filename, which I presume always represents a bug in the mapping?

The approach that mdn-bcd-collector takes to writing BCD is to handle each BCD JSON file individually, first loading it and iterating its entries (BCD paths), making updates, and only writing back to disk if any updates were made. This way you don't have the problem of mapping a BCD path to a filename, because in context you always know what the filename is.